src/cpp/session/include/session/SessionSourceDatabase.hpp

```diff
@@ -188,11 +188,14 @@ core::Error list(std::vector<boost::shared_ptr<SourceDocument> >* pDocs);
 core::Error put(boost::shared_ptr<SourceDocument> pDoc);
 core::Error remove(const std::string& id);
 core::Error removeAll();
+core::Error getPath(const std::string& id, std::string* pPath);
 
 // source database events
 struct Events : boost::noncopyable
 {
    boost::signal<void(boost::shared_ptr<SourceDocument>)> onDocUpdated;
+   boost::signal<void(const std::string&,
```

Comment:
Do the other uses of these events need to worry about onDocRenamed as well (i.e. is it purely additive or does it change the behavior of any of the other events).